lamindb/_curate.py

```diff
@@ -1058,6 +1062,490 @@ def save_artifact(
         return self._artifact
 
 
+def _maybe_validation_error(nonval_keys: list[str], name: str):
```

Comment:
```suggestion
def _maybe_validation_error(nonval_keys: list[str], name: str) -> None:
```

I don't really like the name of this function because "validation" is extremely generic and overloaded heavily in this context. It's more like "maybe_curation_keys_error" or something like this.